File: tests/fixtures/sample_project_java/src/main/java/com/example/auth/Handlers.java
package com.example.auth;

import com.example.db.Connection;
import com.example.users.Crud;
import com.example.users.User;

/** Authentication request handlers — login and session verification. */
public class Handlers {
    /** Validate credentials and issue an access token. */
    public static String authenticateUser(Connection conn, String email) {
        User user = Crud.getUserByEmail(conn, email);
        return JwtUtils.encodeToken(user.email);
    }

    /** Return true if the access token is currently valid. */
    public static boolean verifySession(String token) {
        return JwtUtils.decodeToken(token) != null;
    }
}

File: tests/fixtures/sample_project_rust/src/auth/handlers.rs
//! Authentication request handlers — login and session verification.

use crate::auth::jwt_utils::{decode_token, encode_token};
use crate::db::connection::Connection;
use crate::users::crud::{get_user_by_email, User};

/// Validate credentials and issue an access token.
pub fn authenticate_user(conn: &Connection, email: &str) -> Option<String> {
    let user: User = get_user_by_email(conn, email)?;
    Some(encode_token(&user.email))
}

/// Return true if the access token is currently valid.
pub fn verify_session(token: &str) -> bool {
    decode_token(token).is_some()
}
